### ISSUE TITLE
UselessParenthesesSniff: false positive with xor condition in ternary operator

### ISSUE THREAD

**Issue by schlndh** [slevomat/coding-standard#1418]
If I have an expression like this: `(0 xor 1) ? 0 : 1` `UselessParenthesesSniff` complains that the parentheses are useless. However, this is not the case because `xor` has a [lower precedence](https://www.php.net/manual/en/language.operators.precedence.php) than ternary operator. Note that the same issue also happens with the other operators with lower precedence than ternary operator (other than `or` and `and` which are already handled). [Here](https://3v4l.org/0YmWt) is a quick comparison.

I tried adding these lines to `uselessParenthesesNoErrors.php` and other than `or` and `=` they are all reported as errors.

```php
$output = (($a or $b) ? 'yes' : 'nope');
$output = (($a xor $b) ? 'yes' : 'nope');
$output = ((print 1) ? 'yes' : 'nope');
$output = (($a = 1) ? 'yes' : 'nope');
$output = (($a += 1) ? 'yes' : 'nope');
$output = (($a -= 1) ? 'yes' : 'nope');
$output = (($a *= 1) ? 'yes' : 'nope');
$output = (($a **= 1) ? 'yes' : 'nope');
$output = (($a /= 1) ? 'yes' : 'nope');
$output = (($a .= 1) ? 'yes' : 'nope');
$output = (($a %= 1) ? 'yes' : 'nope');
$output = (($a &= 1) ? 'yes' : 'nope');
$output = (($a |= 1) ? 'yes' : 'nope');
$output = (($a ^= 1) ? 'yes' : 'nope');
$output = (($a <<= 1) ? 'yes' : 'nope');
$output = (($a >>= 1) ? 'yes' : 'nope');
$output = (($a ??= 1) ? 'yes' : 'nope');
```

I was going to skip these the same way that `and` and `or` are skipped ([here](https://github.com/slevomat/coding-standard/blob/94a936419da77f774095aa924b5c4a18f733c3a4/SlevomatCodingStandard/Sniffs/PHP/UselessParenthesesSniff.php#L216)). But then I realized that there could still be useless parentheses inside: e.g. `(($a xor $b) xor $c) ? 0 : 1`. Since `xor` is left-associative `($a xor $b)` doesn't need the parentheses.

So maybe a more complex approach would be in order. On the other hand, even the simple fix is a lot better than nothing (i.e. it's better to ignore some useless parentheses than remove necessary ones).

**Comment by kukulich** [comment]
Thanks. I think, this is good enough: https://github.com/slevomat/coding-standard/commit/eb48bcbddb57f1ae7fe52be29b5beb4fbf77d1cd
